### PR TITLE
Add Oathkeeper logs in integration tests

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Add logs for Oathkeeper pods and container logs when APIGateway CR is in unexpected state during integration tests

**Related issues**

### CODE REVIEW

File: tests/integration/pkg/oathkeeper/log.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	})
+	if err != nil {
+		log.Printf("Fetching Oathkeeper pods for logging: %s", err.Error())
+	}
+
+	for _, pod := range pods.Items {
+
+		log.Printf("Pod %s status: %s", pod.Name, pod.Status.Phase)
+		for _, condition := range pod.Status.Conditions {
+			log.Printf("Pod %s condition: '%s', status: '%s', reasson: '%s' message: '%s'", pod.Name,
```
Comment (barchw) [inline tests/integration/pkg/oathkeeper/log.go:31]:
```suggestion
			log.Printf("Pod %s condition: '%s', status: '%s', reason: '%s' message: '%s'", pod.Name,
```

File: tests/integration/pkg/oathkeeper/log.go
Comment (barchw) [inline tests/integration/pkg/oathkeeper/log.go:1]: Could you add a prefix to the logger (e.g. container prefix)? Right now the logs look little bit chaotic, example:

```
2024/11/28 13:32:21 Logs for container init:
2024/11/28 13:32:21 
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx status: Running
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx condition: 'Initialized', status: 'True', reasson: '' message: ''
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx condition: 'Ready', status: 'False', reasson: 'ContainersNotReady' message: 'containers with unready status: [istio-proxy oathkeeper]'
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx condition: 'ContainersReady', status: 'False', reasson: 'ContainersNotReady' message: 'containers with unready status: [istio-proxy oathkeeper]'
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx condition: 'PodScheduled', status: 'True', reasson: '' message: ''
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx container istio-proxy status: {nil &ContainerStateRunning{StartedAt:2024-11-28 13:32:15 +0100 CET,} nil}
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx container oathkeeper status: {nil &ContainerStateRunning{StartedAt:2024-11-28 13:32:16 +0100 CET,} nil}
2024/11/28 13:32:21 Pod ory-oathkeeper-54bfbc949b-6d6qx container oathkeeper-maester status: {nil &ContainerStateRunning{StartedAt:2024-11-28 13:32:16 +0100 CET,} nil}
2024/11/28 13:32:21 Logs for container oathkeeper:
2024/11/28 13:32:21 time=2024-11-28T12:32:16Z level=info msg=Config file loaded successfully. audience=application path=/etc/config/config.yaml service_name=oathkeeper service_version=
Thank you for using ORY Oathkeeper v0.38.25-beta.1!

Take security seriously and subscribe to the ORY Security Newsletter. Stay on top of new patches and security insights.                                                                                                

>> Subscribe now: http://eepurl.com/di390P <<
time=2024-11-28T12:32:16Z level=info msg=Software quality assurance features are enabled. Learn more at: https://www.ory.sh/docs/ecosystem/sqa audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
time=2024-11-28T12:32:16Z level=info msg=No tracer configured - skipping tracing setup audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
time=2024-11-28T12:32:16Z level=info msg=Listening on http://:9000 audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
time=2024-11-28T12:32:16Z level=info msg=TLS has not been configured for api, skipping audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
time=2024-11-28T12:32:16Z level=info msg=Listening on http://:4456 audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
time=2024-11-28T12:32:16Z level=info msg=TLS has not been configured for proxy, skipping audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
time=2024-11-28T12:32:16Z level=info msg=Listening on http://:4455 audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1

2024/11/28 13:32:21 Logs for container oathkeeper-maester:
2024/11/28 13:32:21 2024-11-28T12:32:16.524Z    INFO    setup   running in sidecar mode
2024-11-28T12:32:17.242Z        INFO    controller-runtime.metrics      metrics server is starting to listen    {"addr": "0.0.0.0:8080"}
2024-11-28T12:32:17.245Z        INFO    setup   using default values for authenticatorsAvailable
2024-11-28T12:32:17.245Z        INFO    setup   using default values for authorizersAvailable
2024-11-28T12:32:17.246Z        INFO    setup   starting manager
2024-11-28T12:32:17.248Z        INFO    controller-runtime.manager      starting metrics server {"path": "/metrics"}
2024-11-28T12:32:17.251Z        INFO    controller-runtime.manager.controller.rule      Starting EventSource    {"reconciler group": "oathkeeper.ory.sh", "reconciler kind": "Rule", "source": "kind source: /, Kind="}
2024-11-28T12:32:17.356Z        INFO    controller-runtime.manager.controller.rule      Starting Controller     {"reconciler group": "oathkeeper.ory.sh", "reconciler kind": "Rule"}
2024-11-28T12:32:17.357Z        INFO    controller-runtime.manager.controller.rule      Starting workers        {"reconciler group": "oathkeeper.ory.sh", "reconciler kind": "Rule", "worker count": 1}

```

**Review by barchw — CHANGES_REQUESTED** [review]

File: tests/integration/pkg/oathkeeper/log.go
Comment (barchw) [reply tests/integration/pkg/oathkeeper/log.go:1]: It might also make sense to save those to a file

Comment (triffer) [reply tests/integration/pkg/oathkeeper/log.go:1]: I added separate loggers for each container and the general logger.
Output:
```oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz status: Running
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz condition: 'PodReadyToStartContainers', status: 'True', reason: '' message: ''
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz condition: 'Initialized', status: 'True', reason: '' message: ''
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz condition: 'Ready', status: 'True', reason: '' message: ''
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz condition: 'ContainersReady', status: 'True', reason: '' message: ''
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz condition: 'PodScheduled', status: 'True', reason: '' message: ''
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz container istio-proxy status: {nil &ContainerStateRunning{StartedAt:2024-11-28 14:28:23 +0100 CET,} nil}
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz container oathkeeper status: {nil &ContainerStateRunning{StartedAt:2024-11-28 14:28:24 +0100 CET,} nil}
oathkeeper-log 2024/11/28 14:29:30 Pod ory-oathkeeper-68cbcd7bcd-lzczz container oathkeeper-maester status: {nil &ContainerStateRunning{StartedAt:2024-11-28 14:28:24 +0100 CET,} nil}
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=Config file loaded successfully. audience=application path=/etc/config/config.yaml service_name=oathkeeper service_version=
oathkeeper-container 2024/11/28 14:29:30 Thank you for using ORY Oathkeeper v0.38.25-beta.1!
oathkeeper-container 2024/11/28 14:29:30 
oathkeeper-container 2024/11/28 14:29:30 Take security seriously and subscribe to the ORY Security Newsletter. Stay on top of new patches and security insights.                                                                                                
oathkeeper-container 2024/11/28 14:29:30 
oathkeeper-container 2024/11/28 14:29:30 >> Subscribe now: http://eepurl.com/di390P <<
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=Software quality assurance features are enabled. Learn more at: https://www.ory.sh/docs/ecosystem/sqa audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=No tracer configured - skipping tracing setup audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=Listening on http://:9000 audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=TLS has not been configured for api, skipping audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=Listening on http://:4456 audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=TLS has not been configured for proxy, skipping audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-container 2024/11/28 14:29:30 time=2024-11-28T13:28:24Z level=info msg=Listening on http://:4455 audience=application service_name=ORY Oathkeeper service_version=v0.38.25-beta.1
oathkeeper-maester-container 2024/11/28 14:29:30 2024-11-28T13:28:24.399Z       INFO    setup   running in sidecar mode
oathkeeper-maester-container 2024/11/28 14:29:30 2024-11-28T13:28:25.092Z       INFO    controller-runtime.metrics      metrics server is starting to listen    {"addr": "0.0.0.0:8080"}
oathkeeper-maester-container 2024/11/28 14:29:30 2024-11-28T13:28:25.095Z       INFO    setup   using default values for authenticatorsAvailable
oathkeeper-maester-container 2024/11/28 14:29:30 2024-11-28T13:28:25.095Z       INFO    setup   using default values for authorizersAvailable
oathkeeper-maester-container 2024/11/28 14:29:30 2024-11-28T13:28:25.097Z       INFO    setup   starting manager
```

Comment (triffer) [reply tests/integration/pkg/oathkeeper/log.go:1]: About saving it to a file ... I'm not sure about it. I hope this logging will only stay until we have more details and can resolve the error. Therefore I don't think it's worth the additional effort in the workflows.

**Pull request merged**